### PR TITLE
ThemeData.copyWith() should not have a  primarySwatch parameter

### LINKED ISSUE #7122: ThemeData.copyWith missing primarySwatch
final myTheme = Theme.of(context).copyWith(primarySwatch: Colors.blue);

myTheme will still use the default primarySwatch. Looking at the copyWith implementation, it appears to simply be missing any logic to copy that parameter (though the param is specified in the signature)

### PR DESCRIPTION
**Pull request by HansMuller**
Fixes #7122

### CODE REVIEW

File: packages/flutter/lib/src/material/theme_data.dart [hunk 1]
```diff
@@ -395,6 +395,7 @@ class ThemeData {
   }) {
     return new ThemeData(
       brightness: brightness ?? this.brightness,
+      primarySwatch: primarySwatch,
```
Comment (abarth) [inline packages/flutter/lib/src/material/theme_data.dart]: Is there no `this.primarySwatch` ?

Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: Aha! As we've just seen a ThemeData object cannot have a null property and we assert as much. So primarySwatch, which is only used to provide default values for unspecified ThemeData constructor parameters, shouldn't be a parameter to ThemeData.copyWith() at all.

File: packages/flutter/lib/src/material/theme_data.dart [hunk 2]
```diff
@@ -393,9 +392,11 @@ class ThemeData {
     IconThemeData accentIconTheme,
     TargetPlatform platform,
   }) {
+    // ThemeData objects are constructed by ThemeData.raw() and do not have
+    // any null-valued properties. All of the constructor parameters below
+    // will also be non-null and its why a primarySwatch parameter isn't needed.
```
Comment (Hixie) [inline packages/flutter/lib/src/material/theme_data.dart]: why is it using the regular constructor and not .raw then?

Comment (HansMuller) [inline packages/flutter/lib/src/material/theme_data.dart]: I do not know why it was written that way. 

I've replaced the comment and the ordinary ThemeData constructor with ThemeData.raw.

**Comment by Hixie** [comment]
LGTM

**Pull request merged**